pkg/hubble/metrics/api/context.go

```diff
@@ -524,3 +525,26 @@ func (o *ContextOptions) Status() string {
 
 	return strings.Join(status, ",")
 }
+
+func (o *ContextOptions) DeleteMetricsAssociatedWithPod(name string, namespace string, vec *prometheus.MetricVec) {
+
```

Comment:
We probably want to preallocate the `sourceCtxCfg` and `destinationCtxCfg`, 

```suggestion
sourceCtxCfg := make([]type, 0, len(o.SourceEgress) + len(o.SourceIngress))
```

Reply:
refactored to calculate it once on context initialization instead of every PodDeletion handling